go/vt/sqlparser/sql.y

```diff
@@ -638,6 +645,93 @@ create_statement:
   {
     $$ = &DBDDL{Action: CreateStr, DBName: string($4)}
   }
+| CREATE definer_opt TRIGGER ID trigger_time trigger_event ON table_name FOR EACH ROW trigger_order_opt lexer_position trigger_body lexer_position
+  {
+    $$ = &DDL{Action: CreateStr, Table: $8, TriggerSpec: &TriggerSpec{Name: string($4), Time: $5, Event: $6, Order: $12, Body: $14}}
+  }
+
+definer_opt:
+  {
+    $$ = struct{}{}
+  }
+| DEFINER '=' ID
+  {
+    $$ = struct{}{}
+  }
+
+trigger_time:
+  BEFORE
+  {
+    $$ = BeforeStr
+  }
+| AFTER
+  {
+    $$ = AfterStr
+  }
+
+trigger_event:
+  INSERT
+  {
+    $$ = InsertStr
+  }
+| UPDATE
+  {
+    $$ = UpdateStr
+  }
+| DELETE
+  {
+    $$ = DeleteStr
+  }
+
+trigger_order_opt:
+  {
+    $$ = nil
+  }
+| FOLLOWS ID
+  {
+    $$ = &TriggerOrder{PrecedesOrFollows: FollowsStr, OtherTriggerName: string($2)}
+  }
+| PRECEDES ID
+  {
+    $$ = &TriggerOrder{PrecedesOrFollows: PrecedesStr, OtherTriggerName: string($2)}
+  }
+
+trigger_body:
+  begin_end_block
+  {
+    $$ = $1
+  }
+| set_statement
+| insert_statement
+| update_statement
+| delete_statement
+
+begin_end_block:
+  BEGIN statement_list ';' END
```

Comment:
What about an empty `BEGIN END` block?

Reply:
Technically allowed but dumb. I'll come back to it.